openpype/pipeline/publish/lib.py

```diff
@@ -866,3 +868,33 @@ def add_repre_files_for_cleanup(instance, repre):
     for file_name in files:
         expected_file = os.path.join(staging_dir, file_name)
         instance.context.data["cleanupFullPaths"].append(expected_file)
+
+
+def get_publish_instance_label(instance, default=_ARG_PLACEHOLDER):
+    """Try to get label from pyblish instance.
+
+    First are checked 'label' and 'name' keys in instance data. If are not set
+    a default value is returned. Instance object is converted to string
+    if default value is not specific.
+
+    Todos:
+        Maybe 'subset' key could be used too.
+
+    Args:
+        instance (pyblish.api.Instance): Pyblish instance.
+        default (Optional[Any]): Default value to return if any
+
+    Returns:
+        Union[Any]: Instance label or default label.
+    """
+
+    label = (
+        instance.data.get("label")
+        or instance.data.get("name")
+    )
+    if label:
+        return label
+
+    if default is _ARG_PLACEHOLDER:
+        return str(instance)
+    return default
```

Comment:
Why can't default just be `None`
```suggestion
def get_publish_instance_label(instance, default=None):
    """Try to get label from pyblish instance.

    First are checked 'label' and 'name' keys in instance data. If are not set
    a default value is returned. Instance object is converted to string
    if default value is not specific.

    Todos:
        Maybe 'subset' key could be used too.

    Args:
        instance (pyblish.api.Instance): Pyblish instance.
        default (Optional[Any]): Default value to return if any

    Returns:
        Union[Any]: Instance label or default label.
    """

    label = (
        instance.data.get("label")
        or instance.data.get("name")
    )
    if label:
        return label

    if default is None:
        return str(instance)
    return default
```

Reply:
Wouldn't you always want it to return a string anyway?

Reply:
For most of cases yes, but I've used different default for debugging for example. Is it a harm to have this way?

Reply:
Not huge harm - just more code complexity for something that currently in the code base is unused too. I'd even go as far as saying that currently the `default=` fallback is redundant since it's unused currently and thus there has not been a use case where we _required_ to deviate from just returning `instance._name` (which is what `str(instance)` does.)

Reply:
Didn't realize `name` is required argument, added the change to PR https://github.com/ynput/OpenPype/pull/5034 here.